src/client/stdlib/generators/input.js

```diff
@@ -4,7 +4,7 @@ export function input(element) {
   return observe((change) => {
     const event = eventof(element);
     let value = valueof(element);
-    const inputted = () => change(valueof(element));
+    const inputted = (event) => change(valueof(event.target));
```

Comment:
The fact that the initial value is `valueof(element)` while subsequent values are `valueof(event.target)` suggests that this may not be desirable. I see the benefit of being able to listen to a parent element, but in that case, how to determine the initial value of the generator? The generator would need to search within the container for an element that exposes a value. (And pick the first of multiple, arbitrarily?)

We could have the `resize` helper proxy the `value` of the returned element. Then you could say:

````md
```js
const chart = resize((width) => scatterPlot(width));
const value = Generators.input(chart);
```

<div class="grid grid-cols-1">
  <div class="card">${chart}</div>
</div>
````

Reply:
Good remark, I hadn't thought about the initial value. I think I'd prefer the “search” approach because it would be easier to use — and also because it fits my mental model of an event that bubbles up (wants to be seen and useful).